src/core/stream/representation/get_needed_segments.ts

```diff
@@ -246,6 +246,22 @@ export default function getNeededSegments({
       }
     }
 
+    // check if the browser is not just garbage collecting it
+    const segmentHistory = getBufferedHistory(contentObject);
+    if (segmentHistory.length > 1) {
+      const lastTimeItWasPushed = segmentHistory[segmentHistory.length - 1];
+      const beforeLastTimeItWasPushed = segmentHistory[segmentHistory.length - 2];
+      if (lastTimeItWasPushed.buffered === null &&
+          beforeLastTimeItWasPushed.buffered === null
+      ) {
+        log.warn("Stream: Segment GCed each time, ignoring it",
+                 adaptation.type,
+                 representation.id,
+                 segment.time);
+        return false;
```

Comment:
In the case of a small `wantedBufferAhead` and not a lot of memory available on the device, can't we fall in a situation where the browser GC the next segment, then we skip it, thus having "holes" in our video ?

Reply:
I wrote about the two strategies because there is a much higher risk of the browser GCing due to being low-in-memory when segments are far from the current position, as it is how most browsers' garbage collection algorithm work

Reply:
I did tests but I'm unsure that they are representative.

When exceeding the point where GC is triggered, the browser actually GCed a lot more than just the last pushed segment, at some random location.
The RxPlayer would there re-download the GCed data until re-exceeding that point and re-download things again.

Reply:
maybe instead of just tracking if the segment has already been GCed , we could track how many times the GC on that segment occured and then we introduce a bias depending on how far the segment is from the current position and skipping it based on a probability 

like a segment that is very far ahead and already have been GC 4/5 times, then it's probably best to skip it ( with like a 90% probability ) but a close segment that only have been GCed once, maybe we shouldn't skip it ( with like a 5% probability to skip ) 

It certainly affect the predictibility of the player, but on average it'll be better 
What do you think about this solution ?

Reply:
Yes we might try that, though this means a lot more logic and complexity.

What if the segment was far the first time and close the second time ? Maybe we should compare with the furthest position of each sample size?

Also it would mean saving the segment distance somewhere, I imagine in the history? For now, this would not really be straightforward so I have to think about how to implement it

Reply:
Also, it might appear a little random as if it fails 4 times due to classic GCing and we are paused, what is preventing the GC from removing that segment a fifth time?